test/test.ml

```diff
@@ -196,6 +212,28 @@ let create_vm id =
 
 let sl x = Printf.sprintf "[ %s ]" (String.concat "; " (List.map (fun (k, v) -> k ^ ":" ^ v) x))
 
+let rpc_of_action = Rpcmarshal.marshal Vm.action.Rpc.Types.ty
+let rpc_of_video_card = Rpcmarshal.marshal Vm.video_card.Rpc.Types.ty
+
+type disk_list = Xenops_types.TopLevel.disk list [@@deriving rpcty]
+
+let rpc_of_disk = Rpcmarshal.marshal Xenops_types.TopLevel.disk.Rpc.Types.ty
+let rpc_of_disk_list = Rpcmarshal.marshal disk_list.Rpc.Types.ty
```

Comment:
Don't you get the marshalling functions 'free' with the `deriving rpcty` addition?

Reply:
This is what you get from deriving, there is a description of the type (`typ_of` and `t`) that can be used with the generic marshaling function:
```
type t = int
val typ_of : int Rpc.Types.typ = Rpc.Types.Basic Rpc.Types.Int
val t : int Rpc.Types.def =
  {Rpc.Types.name = "t"; description = []; ty = Rpc.Types.Basic Rpc.Types.Int
```